drivers/backup/portworx/portworx.go

```diff
@@ -1092,6 +1097,314 @@ func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (
+		Interval: &api.SchedulePolicyInfo_IntervalPolicy{
+			Retain:  retain,
+			Minutes: min,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateDailySchedulePolicy(retain int64, time string, incrCount uint64) *api.SchedulePolicyInfo {
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Daily: &api.SchedulePolicyInfo_DailyPolicy{
+			Retain: retain,
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateWeeklySchedulePolicy(retain int64, day backup.Weekday, time string, incrCount uint64) *api.SchedulePolicyInfo {
+
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Weekly: &api.SchedulePolicyInfo_WeeklyPolicy{
+			Retain: retain,
+			Day:    string(day),
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateMonthlySchedulePolicy(retain int64, date int64, time string, incrCount uint64) *api.SchedulePolicyInfo {
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Monthly: &api.SchedulePolicyInfo_MonthlyPolicy{
+			Retain: retain,
+			Date:   date,
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) BackupSchedulePolicy(name string, uid string, orgId string, schedulePolicyInfo *api.SchedulePolicyInfo) error {
+	log.InfoD("Create Backup Schedule Policy")
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		err = fmt.Errorf("Failed to fetch px-central-admin ctx: [%v]", err)
+		return err
+	}
+	schedulePolicyCreateRequest := &api.SchedulePolicyCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  name,
+			Uid:   uid,
+			OrgId: orgId,
+		},
+		SchedulePolicy: schedulePolicyInfo,
+	}
+	_, err = p.CreateSchedulePolicy(ctx, schedulePolicyCreateRequest)
+	if err != nil {
+		err = fmt.Errorf("Error in creating schedule policy is +%v", err)
+		return err
+	}
+	return nil
+}
+
+func (p *portworx) RegisterBackupCluster(orgID, clusterName, uid string) (api.ClusterInfo_StatusInfo_Status, string) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Failed to fetch px-central-admin ctx")
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true}
+	clusterResp, err := p.InspectCluster(ctx, clusterReq)
+	log.FailOnError(err, "Cluster Object for cluster %s and Org id %s is empty", clusterName, orgID)
+	clusterObj := clusterResp.GetCluster()
+	return clusterObj.Status.Status, clusterObj.Uid
+}
+
+func (p *portworx) GetRuleUid(orgID string, ctx context.Context, ruleName string) (string, error) {
+	RuleEnumerateReq := &api.RuleEnumerateRequest{
+		OrgId: orgID,
+	}
+	rule_list, err := p.EnumerateRule(ctx, RuleEnumerateReq)
```

Comment:
Change `rule_list ` to camel case everywhere

Reply:
Same with `bkp_pod` , `container_list` and `bkp_pods`

Reply:
done!!